Documentation/openshift.md

```diff
@@ -0,0 +1,101 @@
+---
+title: OpenShift
+weight: 17
+indent: true
+---
+
+# OpenShift
+
+[OpenShift](https://www.openshift.com/) adds a number of security and other enhancements to Kubernetes. In particular, [security context constraints](https://blog.openshift.com/understanding-service-accounts-sccs/) allow the cluster admin to define exactly which permissions are allowed to pods running in the cluster. You will need to define those permissions that allow the Rook pods to run. 
+
+## Rook Privileges
+
+To orchestrate the storage platform, Rook requires the following access in the cluster:
+- Create `hostPath` volumes, for persistence by the Ceph mon and osd pods
+- Run pods in `privileged` mode, for access to `/dev` and `hostPath` volumes
+- Host networking
+
+## Security Context Constraints
+
+Before starting the Rook operator or cluster, create the security context constraints needed by the Rook pods. The following yaml is found in `rook-scc.yaml` under `/cluster/examples/kubernetes`.
+
+```yaml
+kind: SecurityContextConstraints
+apiVersion: v1
+metadata:
+  name: rook
+allowPrivilegedContainer: true
+allowHostNetwork: true
+allowHostDirVolumePlugin: true
+priority: 
+allowedCapabilities: []
+allowHostPorts: false
+allowHostPID: false
+allowHostIPC: false
+readOnlyRootFilesystem: false
+requiredDropCapabilities: []
+defaultAddCapabilities: []
+runAsUser:
+  type: RunAsAny
+seLinuxContext:
+  type: MustRunAs
+fsGroup:
+  type: MustRunAs
+supplementalGroups:
+  type: RunAsAny
+allowedFlexVolumes: 
+  - driver: "rook.io/rook"
+volumes:
+  - configMap
+  - downwardAPI
+  - emptyDir
+  - flexVolume
+  - hostPath
+  - persistentVolumeClaim
+  - projected
+  - secret
+users:
+  # A user needs to be added for each rook service account. 
+  # This assumes running in the default sample "rook" and "rook-system" namespaces.
+  # If other namespaces are configured, the namespaces need to be updated here.
+  - system:serviceaccount:rook-system:rook-agent
+  - system:serviceaccount:rook-system:rook-discover
+  - system:serviceaccount:rook-system:rook-operator
+  - system:serviceaccount:rook:default
+  - system:serviceaccount:rook:rook-ceph-osd
```

Comment:
One motivation for fully implementing the [security design ](https://github.com/rook/rook/blob/master/design/security-model.md) fully implemented was to simplify rook's service accounts and permissions.  Having all these service accounts is unnecessary and complicates PSP policy.